src/xunit.analyzers/AssertEqualGenericShouldNotBeUsedForStringValue.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Xunit.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AssertEqualGenericShouldNotBeUsedForStringValue : AssertUsageAnalyzerBase
+    {
+        private static readonly HashSet<string> EqualMethods = new HashSet<string>(new[] { "Equal", "StrictEqual" });
+
+        public AssertEqualGenericShouldNotBeUsedForStringValue() :
+            base(Constants.Descriptors.X2006_AssertEqualGenericShouldNotBeUsedForStringValue, EqualMethods)
+        {
+        }
+
+        protected override void Analyze(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax invocation, IMethodSymbol method)
+        {
+            if (invocation.ArgumentList.Arguments.Count != 2)
+                return;
+
+            if (!method.IsGenericMethod && method.Name != "StrictEqual")
```

Comment:
Having explained that, I think I should change the check to `!method.IsGenericMethod && method.Name == "Equal"`. That better explains what is being verified.